### ISSUE TITLE
Sync or export/import settings

### ISSUE THREAD

**Issue by ThiefMaster** [insin/tweak-new-twitter#204]
Would be great to quickly get the same settings on all your computers. Sync would be preferred, but event a simple import/export with e.g. JSON to copy and paste would already help.

**Comment by zopieux** [comment]
This doc is 8yo (!) but it says different device types (mobile / desktop) won't get extension settings synced: https://wiki.mozilla.org/Services/Sync/Addon_Sync

**Comment by ThiefMaster** [comment]
Oh, I didn't mean mobile. I meant e.g. between home and work desktop or desktop and laptop.

**Comment by zopieux** [comment]
Yeah, gotcha. I was just settings expectations (assuming the doc is still current).

To the point of this issue, I believe it's a matter of using the correct `storage.sync` API for storing preferences:
https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/storage/sync
https://developer.chrome.com/docs/extensions/reference/storage/#storage-areas

instead of `storage.local` which by design won't be synced across devices:
https://github.com/insin/tweak-new-twitter/blob/master/options.js#L370

Hell, this is probably a 9 byte diff fix. :smile:

**Comment by grand-lotus-iroh** [comment]
+1 for desktop import / export .JSON (or whatever) and optional sync. It would be cool if I could store an export file into my password manager and whenever I play around with new browsers I can more easily use this plugin.

**Comment by insin** [comment]
- [x] Export - currently in the debug options (click the last paragraph in the options 5 times to reveal)
- [x] Import - WIP: https://github.com/insin/control-panel-for-twitter/tree/import-config
- [x] Expose in new options section

**Comment by marlonjames71** [comment]
+1 for this. The experience of applying the settings across multiple browsers is time consuming. Would be much better if we could export the settings and import them on the extension in another browser. Check out how Untrap does it. Something like that would be perfect.

**Comment by biast12** [comment]
+1 for this, would love to just be able to export/import settings, don't need sync if it's too hard to add

**Comment by aagha** [comment]
> * [x]  Export - currently in the debug options (click the last paragraph in the options 5 times to reveal)

Where are the debug options?

**Comment by ostrich** [comment]
Sync would be ideal. Settings export/import would be a good stopgap. Thanks for the great extension.

**Comment by hangukhiphop** [comment]
bump

**Comment by adriangrantdotorg** [comment]
bump

**Comment by insin** [comment]
WIP screenshot:

<img width="906" height="636" alt="Image" src="https://github.com/user-attachments/assets/ac4cc1e0-8923-485f-9821-0fb98b79e2f8" />

**Comment by ErikBuis** [comment]
Any update on this? It seems like the feature is still not implemented in the current version (v4.24.0)

**Comment by Mayurifag** [comment]
I am not related to development of the extension, but it seems, syncing settings will be behind paid subscription if I understand changes right on `v5` branch https://github.com/insin/control-panel-for-twitter/tree/v5 - thats why it seems its not released yet.

I understand, that author wants to get some money on development of extension, but it's still sad, especially that it seems it will be not one-time payment, but a subscription. Lets just hope that subscription will be for using author's service and there will be a free way to import/export json file with settings (i dont even request using my own webdav or clouds like google sync, etc.)

**Comment by ThiefMaster** [comment]
WTF. Not a great idea. Nothing wrong with wanting to make money, but come on, charging for this type of browser extension, and then as a subscription? I'd possibly be fine with a one-time payment to support it - but certainly not a subscription.

**Comment by insin** [comment]
The subscription will only be for use of the sync service we have to provide a server for (a one-off payment is also an option). That has a cost for us, so it's paid, that's also why the App Store version is paid.

The manual export/import being added in v5 is as per the screenshot above, all the paid stuff lives in a separate tab.

**Comment by ThiefMaster** [comment]
Isn't it possible to integrate this with the native Mozilla sync service? 
https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/storage/sync

That way you would not need to operate any infrastructure yourself!

**Comment by insin** [comment]
That's tied to a Firefox account and only works in desktop Firefox. The service we're providing will sync settings across all your browsers on all your devices.

**Comment by ThiefMaster** [comment]
Sure, but probably the vast amount of users will use only Firefox or only Chrome (both of which support this) as their main browser where they use Twitter. So cross-browser sync is probably not needed a lot...
